client.go

```diff
@@ -21,12 +21,12 @@ type GraphQLClient struct {
 	Tracer          opentracing.Tracer
 }
 
-func NewClient() *GraphQLClient {
+func NewClient(maxResponseSize int64) *GraphQLClient {
```

Comment:
I think an optional config object with an `MaxResponseSize` that defaults to `1MB` would be a cleaner interface, then we don't need to provide a default in tests and in the introspection file

Reply:
Yep, while you cannot always truely make backwards incompatible changes due to the signature needing to change. This change would break any existing code importing bramble and calling `bramble.NewClient()` 

I'd recommend something like 

```
type ClientOpt func(*GraphQLClient)

func NewClient(opts ...ClientOpt) *GraphQLClient {
	c := &GraphQLClient{
		HTTPClient: &http.Client{
			Timeout: 5 * time.Second,
		},
		maxResponseSize: 1024 * 1024,
	}
	for _, opt := range opts {
		opt(c)
	}
	return c
}

func WithMaxResponseSize(maxResponseSize int64) ClientOpt {
	return func(s *GraphQLClient) {
		s.maxResponseSize = maxResponseSize
	}
}
```